### DISCUSSION TITLE
Virtual Singers

### DISCUSSION THREAD

**Discussion by JBetz** [free-audio/clap#289] [Ideas]
What would it look like for CLAP to support not just virtual instruments, but also virtual singers? 

The current non-solution is for virtual singer VSTs to have their own embedded timelines (see Emvoice One). In the case of Vocaloid, it's practically an embedded DAW. And since there is no standard mapping to MIDI events, and no representation of lyrics in MIDI, MIDI is forgone entirely in favor of storing the entire state in the VST. This is obviously bad and makes virtual singers a second-class citizen in any and all DAWs, since it's impossible to switch the VST for a virtual singer track the same way you can with instruments.

The simplest solution is to be able to attach words to note events. Or even dipthongs. And either to piggyback on CC messages for expression, or go the NotePerformer route and let the plugin generate a performance with either delayed or offline rendering. And ideally support either option.

Curious if anyone else has thought about this and what designs already exist, if any.

**Comment by jaxter184** [comment]
This was the main motivation behind this post: https://github.com/free-audio/clap/discussions/126
The conclusion there was that a "string" event might make sense. In a voice synth, the formants are not necessarily coupled to the pitch, so it seems reasonable to me that there could be a formant "parameter" that takes arbitrary strings, and a separate note port for the pitch (as opposed to a single message that contains both the formant and the pitch).
Also potentially relevant: https://github.com/free-audio/clap/pull/229#discussion_r1038946788

**Reply by jaxter184** [reply]
Though it's worth noting that triggers and strings aren't very common in most DAWs. But the nature of CLAP's extensions make me feel like it's more likely to end up being useful with CLAP than, say, VST or LV2.
